sklearn/utils/__init__.py

```diff
@@ -58,9 +58,12 @@
 
 
 class Bunch(dict):
-    """Container object for datasets
+    """Container object exposing keys as attributes
+
+    Bunch objects are sometimes used as an output for functions and methods.
+    They extend dictionaries by enabling values to be accessed by key,
+    `bunch["value_key"]`, or by an attribute, `bunch.value_key`.
```

Comment:
Do we want to follow a strict NumPy documentation (add Examples section and something about the Attributes)?

Reply:
Id say that's fine, the docstring is pretty clear